sycl/source/detail/event_impl.hpp

```diff
@@ -192,15 +202,17 @@ class event_impl {
   void *MCommand = nullptr;
 
   /// Dependency events prepared for waiting by backend.
-  std::vector<std::shared_ptr<event_impl>> MPreparedDepsEvents;
-  std::vector<std::shared_ptr<event_impl>> MPreparedHostDepsEvents;
+  std::vector<EventImplPtr> MPreparedDepsEvents;
+  std::vector<EventImplPtr> MPreparedHostDepsEvents;
 
   enum HostEventState : int { HES_NotComplete = 0, HES_Complete };
 
   // State of host event. Employed only for host events and event with no
   // backend's representation (e.g. alloca). Used values are listed in
   // HostEventState enum.
   std::atomic<int> MState;
+
+  std::mutex MMutex;
```

Comment:
Why is this mutex needed? Is access to MPreparedDepsEvents and MPreparedHostDepsEvents not guarded by graph R/W lock?

Reply:
Access to these dependencies is not guarded when the user wants to get them from event::get_wait_list